activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -230,7 +230,15 @@ def column_for_attribute(name)
     #   person.respond_to(:nothing) # => false
     def respond_to?(name, include_private = false)
       return false unless super
-      name = name.to_s
+
+      case name
```

Comment:
Nevermind the string passed is converted to a symbol so we're in the same case probably. So end this probably there is not a "solution" that satisfy performance and readability all of them has some tradeoffs